File: backend/app/controllers/patients.py
from fastapi import APIRouter, Depends, status, Response
from sqlalchemy.orm import Session
from app.core.database import get_db
from app.schemas.patients import PatientResponse, PatientCreate, PatientUpdate
from app.services.patients import PatientService
from typing import List

router = APIRouter(prefix="/patients", tags=["Patients"])

@router.get("/", response_model=List[PatientResponse])
def list_patients(db_session: Session = Depends(get_db)):
    return PatientService.get_all(db_session)

@router.get("/{patient_id}", response_model=PatientResponse)
def get_patient(patient_id: int, db_session: Session = Depends(get_db)):
    return PatientService.get_by_id(db_session, patient_id)

@router.post("/", response_model=PatientResponse, status_code=status.HTTP_201_CREATED)
def create_patient(patient_data: PatientCreate, db_session: Session = Depends(get_db)):
    return PatientService.create(db_session, patient_data)

@router.put("/{patient_id}", response_model=PatientResponse)
def update_patient(patient_id: int, patient_data: PatientUpdate, db_session: Session = Depends(get_db)):
    return PatientService.update(db_session, patient_id, patient_data)

@router.delete("/{patient_id}", status_code=status.HTTP_204_NO_CONTENT)
def delete_patient(patient_id: int, db_session: Session = Depends(get_db)):
    PatientService.delete(db_session, patient_id)
    return Response(status_code=status.HTTP_204_NO_CONTENT)

File: frontend/src/services/patients.ts
import { ApiService } from "./api";
import { Patient } from "@/src/types/patient";

export const PatientService = {
  getAll: () => ApiService.get<Patient[]>("/patients/"),
  
  getById: (id: number) => ApiService.get<Patient>(`/patients/${id}`),
  
  create: (data: Partial<Patient>) => ApiService.post<Patient>("/patients/", data),
  
  update: (id: number, data: Partial<Patient>) => ApiService.put<Patient>(`/patients/${id}`, data),
  
  delete: (id: number) => ApiService.delete(`/patients/${id}`),
};
